src/js/components/dropdown.js

```diff
@@ -0,0 +1,286 @@
+/**
+ * Copyright (C) 2018 The Trustees of Indiana University
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import keyCodes from '../utilities/keyCodes';
+
+export default class Dropdown {
+  constructor(element) {
+    this.element = element;
+    this.focusableElements = 'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]';
+
+    this.dropdownAttribute = '[data-dropdown]';
+
+    this.toggleAttribute = '[data-dropdown-toggle]';
+    this.toggleElement = this.element.querySelector(this.toggleAttribute);
+
+    this.menuAttribute = '[data-dropdown-menu]';
+    this.menuElement = this.element.querySelector(this.menuAttribute);
+
+    // Keeps track of the currently active dropdown and helps with focus management
+    this.activeDropdown = null;
+
+    // Bind methods
+    this._handleClick = this._handleClick.bind(this);
+    this._handleKeydown = this._handleKeydown.bind(this);
+
+    this.init();
+  }
+
+  /**
+   * Opens a dropdown menu
+   * @param {HTMLButtonElement} toggle - a button element which toggles the dropdown
+   * @param {HTMLDivElement} menu - a div containing the dropdown menu items
+   */
+  openMenu(toggle, menu) {
+    const toggleButton = toggle || this.toggleElement;
+    const menuList = menu || this.menuElement;
+
+    // Return if disabled dropdown is being opened programmatically
+    if (toggleButton.hasAttribute('disabled')) {
+      return;
+    }
+
+    // If the menu was opened by clicking an associated toggle
+    if (toggleButton && toggleButton !== null) {
+      toggleButton.setAttribute('aria-expanded', 'true');
+    }
+
+    // Remove the 'hidden' attribute to show the menu
+    menuList.setAttribute('aria-hidden', 'false');
```

Comment:
Let's use the `hidden` attribute instead of `aria-hidden` here since we have been moving that direction in all of the other components. So for example, this would become:

```js
this.menuElement.removeAttribute('hidden');
```